elastic/kibana
# [Monaco editor]: Fix accessibility focus trap issue

fixes https://github.com/elastic/kibana/issues/72677

This PR fixes an accessibility issue with the `<CodeEditor />` component from the `kibana_react` plugin where a user would not be able to exit the editor with a keyboard.

Hitting the ESC key will now display an overlay to the user that allows him to navigate to the next element on the page.

## How to test

* Navigate to the index pattern management app
* Click "Add field"
* Start navigating with the keyboard using the "TAB" key
* Enable "Set value" with the SPACE key
* TAB to access the script editor
* An overlay should appear to indicate to hit "ENTER" to start writing
* Hit ENTER
* Write `emi` ---> you should have the suggestion menu open
* Hit the "ESC" key. The suggestion menu should close
* Hit the "ESC" again ---> you should now see the overlay back
* Hit TAB to go to the next item (the link below the editor)

## Note on implementation

I decided to mock the `MonacoEditor` component in order to be able to not test implementation detail of the `<CodeEditor />` and call methods on the component directly.  This allows us to have an `<textarea />` rendered in our test and interact with it.

![editor_a11y](https://user-images.githubusercontent.com/2854616/128316067-6bdbcf66-8ff5-4026-9ae6-150d437a4af1.gif)
